src/backend/aspen/api/views/groups.py

```diff
@@ -62,17 +64,15 @@ async def get_group_info(
     group_id: int,
     request: Request,
     db: AsyncSession = Depends(get_db),
-    user: User = Depends(get_auth_user),
+    authorized_query: Select = Depends(require_access("read", Group)),
 ) -> GroupInfoResponse:
-    if user.group.id != group_id and not user.system_admin:
-        raise ex.UnauthorizedException("Not authorized")
-    group_query = (
-        sa.select(Group)  # type: ignore
-        .options(joinedload(Group.default_tree_location))
-        .where(Group.id == group_id)
-    )
+    group_query = authorized_query.options(  # type: ignore
+        joinedload(Group.default_tree_location)
+    ).where(Group.id == group_id)
```

Comment:
hrm... then again we join more stuff to these queries so that might not be as awesome as it sounds

Reply:
Yeah. I think SQLAlchemy has methods to populate model attributes like this but I've never played with them, so I don't know how tricky it would be to use them.